cliquet/storage/postgresql/__init__.py

```diff
@@ -586,9 +586,17 @@ def _format_conditions(self, filters, id_field, modified_field,
                 # Safely escape field name
                 field_holder = '%s_field_%s' % (prefix, i)
                 holders[field_holder] = filtr.field
-                # JSON operator ->> retrieves values as text.
-                # If field is missing, we default to ''.
-                sql_field = "coalesce(data->>:%s, '')" % field_holder
+                if filtr.operator in (COMPARISON.LT, COMPARISON.MAX,
+                                      COMPARISON.MIN, COMPARISON.GT):
+                    try:
+                        value = float(value)
+                        sql_field = "data->:%s" % field_holder
+                    except ValueError:
+                        sql_field = "coalesce(data->>:%s, '')" % field_holder
+                else:
+                    # JSON operator ->> retrieves values as text.
+                    # If field is missing, we default to ''.
+                    sql_field = "coalesce(data->>:%s, '')" % field_holder
```

Comment:
Actually it is what I did before and I reverted because doing this ``value = float(value)`` adds problems for phone numbers like ``0123456789`` for instance.